### PR TITLE
As/modify forger

### PR DESCRIPTION
**Pull request by alsala**

### CODE REVIEW

File: sdk/src/main/scala/com/horizen/forge/AbstractForgeMessageBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+      result
+    }
+    case Failure(ex) => {
+      log.error(s"Failed to forge block for ${nextConsensusEpochNumber} epoch ${nextConsensusSlotNumber} slot due:", ex)
+      ForgeFailed(ex)
+    }
+  }
+
+  protected def getSecretsAndProof(
+                     wallet: AbstractWallet[TX, PM, VL],
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/forge/AbstractForgeMessageBuilder.scala:103]: can be `wallet: VL`

File: sdk/src/main/scala/com/horizen/forge/ForgeMessageBuilder.scala [hunk 2]
```diff
@@ -3,210 +3,131 @@ package com.horizen.forge
...
+  mainchainSynchronizer, companion, params, allowNoWebsocketConnectionInRegtest
+) {
+  type HSTOR = SidechainHistoryStorage
+  type VL = SidechainWallet
+  type HIS = SidechainHistory
+  type MS = SidechainState
+  type MP = SidechainMemoryPool
 
-  case class BranchPointInfo(branchPointId: ModifierId, referenceDataToInclude: Seq[MainchainHeaderHash], headersToInclude: Seq[MainchainHeaderHash])
+  type ForgeMessageType = GetDataFromCurrentView[SidechainHistory, SidechainState, SidechainWallet, SidechainMemoryPool, ForgeResult]
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/forge/ForgeMessageBuilder.scala:41]: I think, we can put this and `buildForgeMessageForEpochAndSlot` into the base class as well.
we have all the templates known there.

**Review by i-Alex — CHANGES_REQUESTED** [review]

File: sdk/src/main/scala/com/horizen/forge/Forger.scala [hunk 3]
```diff
@@ -1,171 +1,60 @@
...
-      case failure @ Failure(ex) => {
-        log.error(s"Forging had been failed. Reason: ${ex.getMessage}")
-        respondsToOpt.map(respondsTo => respondsTo ! failure)
-      }
-    }
-  }
+             params: NetworkParams) extends AbstractForger(
+  settings, viewHolderRef, forgeMessageBuilder, timeProvider, params
+) {
+  type View = CurrentView[SidechainHistory, SidechainState, SidechainWallet, SidechainMemoryPool]
```
[the remark this answers is not in the document]
Comment (alsala) [reply sdk/src/main/scala/com/horizen/forge/Forger.scala]: Done

File: sdk/src/main/scala/com/horizen/account/forger/AccountForger.scala [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+  override def getForgedBlockAsFuture(epochNumber: ConsensusEpochNumber, slot: ConsensusSlotNumber, blockCreationTimeout: Timeout) : Future[ForgeResult] = {
+    val forgeMessage: AccountForgeMessageBuilder#ForgeMessageType = forgeMessageBuilder.buildForgeMessageForEpochAndSlot(epochNumber, slot, blockCreationTimeout)
+    val forgedBlockAsFuture = (viewHolderRef ? forgeMessage).asInstanceOf[Future[ForgeResult]]
+    forgedBlockAsFuture
+  }
+}
+
+object AccountForger extends ScorexLogging {
+  object ReceivableMessages {
```
[the remark this answers is not in the document]
Comment (alsala) [reply sdk/src/main/scala/com/horizen/account/forger/AccountForger.scala]: Done

File: sdk/src/main/scala/com/horizen/forge/AbstractForgeMessageBuilder.scala (hunk 1 above)
Comment (alsala) [reply sdk/src/main/scala/com/horizen/forge/AbstractForgeMessageBuilder.scala]: Done

File: sdk/src/main/scala/com/horizen/forge/ForgeMessageBuilder.scala (hunk 2 above)
Comment (alsala) [reply sdk/src/main/scala/com/horizen/forge/ForgeMessageBuilder.scala]: Done

File: sdk/src/main/scala/com/horizen/account/forger/AccountForger.scala [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import com.horizen.params.NetworkParams
+
+import scorex.core.utils.NetworkTimeProvider
+
+
+class AccountForger(settings: SidechainSettings,
+             viewHolderRef: ActorRef,
+             forgeMessageBuilder: AccountForgeMessageBuilder,
+             timeProvider: NetworkTimeProvider,
+             params: NetworkParams) extends AbstractForger(
```
Comment (i-Alex) [inline sdk/src/main/scala/com/horizen/account/forger/AccountForger.scala:21]: `extends AbstractForger[SidechainTypes#SCAT, AccountBlockHeader, AccountBlock](...`
similar for `Forger`

Comment (alsala) [reply sdk/src/main/scala/com/horizen/account/forger/AccountForger.scala]: Done

**Review by i-Alex — APPROVED** [review]

**Pull request merged**